command/resource/read-grpc/read.go

```diff
@@ -0,0 +1,170 @@
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI        cli.Ui
+	flags     *flag.FlagSet
+	grpcFlags *client.GRPCFlags
+	help      string
+
+	filePath string
+}
+
+func (c *cmd) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flags.StringVar(&c.filePath, "f", "",
+		"File path with resource definition")
+
+	c.grpcFlags = &client.GRPCFlags{}
+	client.MergeFlags(c.flags, c.grpcFlags.ClientFlags())
+	c.help = client.Usage(help, c.flags)
+}
+
+func (c *cmd) Run(args []string) int {
+	var resourceType *pbresource.Type
+	var resourceTenancy *pbresource.Tenancy
+	var resourceName string
+
+	if err := c.flags.Parse(args); err != nil {
+		if !errors.Is(err, flag.ErrHelp) {
+			c.UI.Error(fmt.Sprintf("Failed to parse args: %v", err))
+			return 1
+		}
+		c.UI.Error(fmt.Sprintf("Failed to run apply command: %v", err))
+		return 1
+	}
+
+	// collect resource type, name and tenancy
+	if c.flags.Lookup("f").Value.String() != "" {
+		if c.filePath != "" {
+			parsedResource, err := resource.ParseResourceFromFile(c.filePath)
+			if err != nil {
+				c.UI.Error(fmt.Sprintf("Failed to decode resource from input file: %v", err))
+				return 1
+			}
+
+			if parsedResource == nil {
+				c.UI.Error("Unable to parse the file argument")
+				return 1
+			}
+
+			resourceType = parsedResource.Id.Type
+			resourceTenancy = parsedResource.Id.Tenancy
+			resourceName = parsedResource.Id.Name
+		} else {
+			c.UI.Error(fmt.Sprintf("Please provide an input file with resource definition"))
+			return 1
+		}
+	} else {
+		var err error
+		resourceType, resourceName, err = resource.GetTypeAndResourceName(args)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Incorrect argument format: %s", err))
+			return 1
+		}
+
+		inputArgs := args[2:]
+		err = resource.ParseInputParams(inputArgs, c.flags)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error parsing input arguments: %v", err))
+			return 1
+		}
+		if c.filePath != "" {
+			c.UI.Error("Incorrect argument format: File argument is not needed when resource information is provided with the command")
+			return 1
+		}
+		resourceTenancy = &pbresource.Tenancy{
+			Namespace: c.grpcFlags.Namespace(),
+			Partition: c.grpcFlags.Partition(),
+			PeerName:  c.grpcFlags.Peername(),
+		}
+	}
+
+	// initialize client
+	config, err := client.LoadGRPCConfig(nil)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error loading config: %s", err))
+		return 1
+	}
+	c.grpcFlags.MergeFlagsIntoGRPCConfig(config)
+	resourceClient, err := client.NewGRPCClient(config)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error connect to Consul agent: %s", err))
+		return 1
+	}
+
+	// read resource
+	res := resource.ResourceGRPC{C: resourceClient}
+	entry, err := res.Read(resourceType, resourceTenancy, resourceName, c.grpcFlags.Stale())
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading resource %s/%s: %v", resourceType, resourceName, err))
+		return 1
+	}
+
+	// display response
+	b, err := json.MarshalIndent(entry, "", "    ")
```

Comment:
Did you get a chance to see if the indentation is consistent across the entire cli? Maybe we can switch to using a constant instead

Reply:
yup, moved it to as a constant